pyccel/codegen/printing/cwrappercode.py

```diff
@@ -518,7 +687,11 @@ def _print_Module(self, expr):
             static_funcs = expr.funcs
         function_signatures = '\n'.join('{};'.format(self.function_signature(f)) for f in static_funcs)
 
-        function_defs = '\n\n'.join(self._print(f) for f in expr.funcs)
+        #TODO it should be a better way to do this :
+        funcs = expr.interfaces
+        funcs += [f for f in expr.funcs if f.name not in [item.name for s in funcs for item in s.functions]]
```

Comment:
I would split this into 2 lines to make sure that python doesn't try to create the inner list on every loop. So:
```
item_names = [item.name for s in funcs for item in s.functions]
funcs += [f for f in expr.funcs if f.name not in item_names]
```

What is item_names?

Reply:
funcs = list of interfaces
s.functions = function inside interfaces 
item = function
item.name = function.name

Reply:
:+1:
I find the naming confusing. Is this equivalent to:
```
interfaces = expr.interfaces
interface_funcs = [f.name for i in interfaces for f in i.functions]
funcs = interfaces + [f for f in expr.funcs if f.name not in interface_funcs]
```
?

Reply:
yes it s equivalent

Reply:
Then would you mind changing the names to something similar to make this clearer

Reply:
Thanks